File: 1064. 不动点/1064.java
class Solution {
    public int fixedPoint(int[] A) {
        int left = 0;
        int right = A.length - 1;
        int mid = 0;
        while (left < right) {
            mid = (left + right) / 2;
            if (A[mid] < mid) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        return A[left] == left ? left : -1;
    }
}

File: 1064. 不动点/1064.rs
impl Solution {
    pub fn fixed_point(arr: Vec<i32>) -> i32 {
        let mut left = 0;
        let mut right = arr.len() - 1;
        while (left < right) {
            let mid = (left + right) / 2;
            if (arr[mid] < mid as i32) {
                left = mid + 1;
            } else {
                right = mid;
            }
        }
        if arr[left] == left as i32{
            return left as i32;
        }
        else{
            return -1;
        }
    }
}